File: lib/app/application.cc
// Copyright 2016 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include <magenta/processargs.h>
#include <mxio/util.h>

#include "apps/modular/lib/app/application.h"

namespace modular {

Application::Application(
    fidl::InterfaceHandle<ServiceProvider> environment_services,
    fidl::InterfaceRequest<ServiceProvider> outgoing_services)
    : environment_services_(
          ServiceProviderPtr::Create(std::move(environment_services))),
      outgoing_services_(std::move(outgoing_services)) {}

Application::~Application() = default;

std::unique_ptr<Application> Application::CreateFromStartupInfo() {
  mx_handle_t incoming = mxio_get_startup_handle(MX_HND_TYPE_INCOMING_SERVICES);
  mx_handle_t outgoing = mxio_get_startup_handle(MX_HND_TYPE_OUTGOING_SERVICES);

  return std::make_unique<Application>(
      fidl::InterfaceHandle<ServiceProvider>(mx::channel(incoming), 0u),
      fidl::InterfaceRequest<ServiceProvider>(mx::channel(outgoing)));
}

}  // namespace modular

File: public/lib/fidl/dart/application.dart
// Copyright 2014 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

library application;

import 'dart:async';
import 'dart:collection';

import 'package:lib.fidl.dart/bindings.dart' as bindings;
import 'package:lib.fidl.dart/core.dart' as core;
import 'package:mojo.public.interfaces.application/application.mojom.dart' as application_mojom;
import 'package:mojo.public.interfaces.application/service_provider.mojom.dart';
import 'package:mojo.public.interfaces.application/shell.mojom.dart' as shell_mojom;
import 'package:mojo.public.interfaces.bindings/service_describer.mojom.dart'
    as service_describer;

part 'src/application.dart';
part 'src/application_connection.dart';
part 'src/service_describer.dart';
